src/main/java/dinkplugin/util/ItemSearcher.java

```diff
@@ -0,0 +1,119 @@
+package dinkplugin.util;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Singleton
+public class ItemSearcher {
+    private final Map<String, Integer> itemIdByName = Collections.synchronizedMap(new HashMap<>());
+    private @Inject OkHttpClient httpClient;
+    private @Inject Gson gson;
+
+    @Nullable
+    public Integer findItemId(String name) {
+        return itemIdByName.get(name);
```

Comment:
Felan brought up that `Chompy bird hat` would map to a single ID as there are [multiple variations](https://oldschool.runescape.wiki/w/Collection_log#Chompy_Birds) that all have the same in-game name. 

@pajlada Should we inspect the player's inventory to select the correct ID or not bother? (I can implement this with Guava's MultiMap)

Reply:
If this happens to items that are more rare or fancy than the Chompy bird hat then I think we should take effort to fix it, otherwise leave it.